### PR TITLE
[#4] [UI] As a user, I can see My Coins on the Home screen

### LINKED ISSUE #4: [UI] As a user, I can see My Coins on the Home screen
## Why

- The user should be able to see their coins on the Home screen.

## Acceptance Criteria

- On the Home screen, the user can see the following My Coins section:

_In light mode:_

Home Page

_In dark mode:_

Home Page

- The section will have the title `My Coins 😎`.
- The section will have the text `see all ->`.

- Once the coins have loaded, the user will see coin’s items:

Bitcoin

- The coin’s image will be based on the `image`.
- The coin’s symbol will be based on the _uppercased_ `symbol`.
- The coin’s name will be based on the `name`.
- The coin’s current price will be based on the `current_price`.
- The coin’s price change percentage will be based on the `price_change_percentage_24h`:
  - `price_change_percentage_24h` is set to `#F15950` with down indicator when it’s negative number:
Screen Shot 2022-11-23 at 22 26 45
  - `price_change_percentage_24h` is set to `#10DC78` with the up indicator when it’s positive number:
Screen Shot 2022-11-23 at 22 26 48

_Note:_ The fractional part of `price_change_percentage_24h` is rounded to 2 numbers: `6,2122` to `6,21`.

## Design
- [Figma design](https://www.figma.com/file/Iy9eg3ym29pkKGsD5X48O7/Crypto-App-(Community)?node-id=3%3A181&t=W7iFTb0v4cHHYenL-4).

### PR DESCRIPTION
**Pull request by nkhanh44**
- Close #4 

## What happened 👀

- Created MyCoinSection that includes header and MyCoinView.
- Handled dark and light modes based on Assets.

## Insight 📝

In handling dark and light modes:
- For each color (text, title, background), I created a Color set in Assets where I defined specific colors for dark and light modes.

Screenshot 2022-12-12 at 12 09 09

## Proof Of Work 📹

Light mode:

Dark mode:

### CODE REVIEW

File: CryptoPrices/Sources/Home/Package.swift [hunk 1]
```diff
@@ -16,13 +16,15 @@ let package = Package(
...
+        .package(url: "https://github.com/Quick/Nimble", from: "11.2.1"),
+        .package(name: "MyCoin", path: "../MyCoin")
     ],
     targets: [
         .target(
             name: "Home",
             dependencies: [
-                .product(name: "Styleguide", package: "Styleguide")
+                .product(name: "Styleguide", package: "Styleguide"),
+                .product(name: "MyCoin", package: "MyCoin")
```
Comment (Thieurom) [inline CryptoPrices/Sources/Home/Package.swift:27]: The `Home` and `MyCoin` are modules that represent their own features, they belong to the same _horizontal layer_. They shouldn't depend on each other.
Maybe the name `MyCoin` is a little bit confusing and misleading here 🙇  but it was created for the _detail_ screen.
Kindly put all `Home`'s related codes (`MyCoinView`, `MyCoinItemView`) into the `Home` package 🙏

**Review by Thieurom — COMMENTED** [review]
Other than this, the rest looks great to me 👍

File: CryptoPrices/Sources/Home/Package.swift (hunk 1 above)
Comment (nkhanh44) [reply CryptoPrices/Sources/Home/Package.swift]: 🙈 my bad, I thought MyCoin is a module for MyCoins. Fixed it at https://github.com/nimblehq/nimble-crypto-ios/pull/32/commits/5cf90f65644ed6288a7ff72cf0edd66e9b060dc3. 🙏

**Review by Thieurom — APPROVED** [review]
LGTM 🚀

File: CryptoPrices/Sources/Home/Sources/Home/MyCoins/MyCoinItemView.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    
+    var priceChangePercentage: some View {
+        // TODO: - Remove dummy
+        HStack {
+            Image(percentage < 0 ? "ic_arrow_down_red" : "ic_arrow_up_green")
+
+            Text("6.21%")
+                .font(.Inter.medium.font(for: .callout))
+                .foregroundColor(percentage < 0 ? .carnation : .guppieGreen)
```
Comment (minhnimble) [inline CryptoPrices/Sources/Home/Sources/Home/MyCoins/MyCoinItemView.swift:76]: I think we also need to use Float representation for these `percentage` values here 🙏 
```suggestion
            Image(percentage < 0.0 ? "ic_arrow_down_red" : "ic_arrow_up_green")

            Text("6.21%")
                .font(.Inter.medium.font(for: .callout))
                .foregroundColor(percentage < 0.0 ? .carnation : .guppieGreen)
```

File: CryptoPrices/Resources/Assets.xcassets/Colors/bg_currency_item.colorset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        }
+      },
+      "idiom" : "universal"
+    }
+  ],
+  "info" : {
+    "author" : "xcode",
+    "version" : 1
+  }
+}
```
Comment (minhnimble) [inline CryptoPrices/Resources/Assets.xcassets/Colors/bg_currency_item.colorset/Contents.json:56]: Nice usage of assets for handling dark/light color 👍🚀

**Review by minhnimble — COMMENTED** [review]
Just some minor concerns, rest LGTM 🚀

File: CryptoPrices/Sources/Home/Sources/Home/MyCoins/MyCoinItemView.swift (hunk 2 above)
Comment (nkhanh44) [reply CryptoPrices/Sources/Home/Sources/Home/MyCoins/MyCoinItemView.swift]: Nice catch! I fixed it at https://github.com/nimblehq/nimble-crypto-ios/pull/32/commits/5020445330531e4044db489c732a5a16d42af45f. 🙏

File: CryptoPrices/Sources/Home/Sources/Home/MyCoins/MyCoinItemView.swift [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+private extension MyCoinItemView {
+
+    var coinImage: some View {
+        // TODO: - Remove dummy
+        Image("ic_bitcoin")
+            .frame(width: 40.0, height: 40.0)
+            .clipShape(Circle())
+    }
```
[the remark this answers is not in the document]
Comment (nkhanh44) [reply CryptoPrices/Sources/Home/Sources/Home/MyCoins/MyCoinItemView.swift:46]: It's about the style and access control 😄. I just wanted to make all the components of MyCoinItemView's body (`coinImage`, `coinInfo`, `currentPrice`, and `priceChangePercentage`) have a separate private implementation each. The two following image comparisons:

Screenshot 2022-12-14 at 11 22 26
compared with the Image alone.
Screenshot 2022-12-14 at 11 25 05

**Review by minhnimble — APPROVED** [review]

**Pull request merged**